src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Parser/Utf8Parser.Number.cs

```diff
@@ -258,12 +252,37 @@ private static bool TryParseNumber(ReadOnlySpan<byte> source, ref Number.NumberB
                     break;
             }
 
-            if (!Utf8Parser.TryParseUInt32D(source.Slice(srcIndex), out uint absoluteExponent, out int bytesConsumedByExponent))
+            // If the next character isn't a digit, an exponent wasn't specified
+            if ((byte)(c - (byte)('0')) > 9)
             {
                 bytesConsumed = 0;
                 return false;
             }
 
+            if (!TryParseUInt32D(source.Slice(srcIndex), out uint absoluteExponent, out int bytesConsumedByExponent))
+            {
+                // Since we found at least one digit, we know that any failure to parse means we had an
+                // exponent that was larger than uint.MaxValue, and we can just eat characters until the end
+                absoluteExponent = uint.MaxValue;
```

Comment:
So if we find an exponent like 123456789123456789, we round that down to uint.MaxValue and ignore all subsequent digits rather than failing to parse?

Reply:
Correct, parsing doubles and floats requires the input to be parsed to "infinite" precision and then rounded to the nearest representable result.

> Conversion from a character sequence of more than H significant digits or larger in exponent range than the destination binary format first shall be correctly rounded to H digits according to the applicable rounding direction